lms/djangoapps/discussion_api/api.py

```diff
@@ -367,16 +383,19 @@ def get_comment_list(request, thread_id, endorsed, page, page_size):
         discussion_api.views.CommentViewSet for more detail.
     """
     response_skip = page_size * (page - 1)
-    cc_thread, context = _get_thread_and_context(
-        request,
-        thread_id,
-        retrieve_kwargs={
-            "recursive": False,
-            "user_id": request.user.id,
-            "response_skip": response_skip,
-            "response_limit": page_size,
-        }
-    )
+    try:
+        cc_thread, context = _get_thread_and_context(
+            request,
+            thread_id,
+            retrieve_kwargs={
+                "recursive": False,
+                "user_id": request.user.id,
+                "response_skip": response_skip,
+                "response_limit": page_size,
+            }
+        )
+    except Http404:
```

Comment:
Instead of updating all callers of `_get_thread_and_context` , can't we just change `_get_thread_and_context` so it doesn't raise `Http404`?